### PR TITLE
fix: general config updates

### PR DESCRIPTION
**Pull request by raisedadead**

### CODE REVIEW

**Comment by raisedadead** [comment]
To be merged just prior to our upcoming soft launch. Not before that.

**Comment by raisedadead** [comment]
Forgot to mention, I am still working with Chinese NGINX config. We might end-up creating them as their independent config. We are considering to deploy the cluster indepedent of the global cluster closer to china

File: sites-enabled/10-www.freecodecamp.org.conf [hunk 1]
```diff
@@ -116,7 +122,7 @@ server {
   # SSL
   include snippets/common/ssl-freecodecamp-org.conf;
 
-  return 301 https://www.freecodecamp.org$request_uri;
+  return 302 https://www.freecodecamp.org$request_uri;
```
[the remark this answers is not in the document]
Comment (raisedadead) [reply sites-enabled/10-www.freecodecamp.org.conf:125]: Good question. 

In this case, the server block is for the host `.freecodecamp.org`. This is a catch all for, and superseded when we add something on a subdomain. 

While we well almost never have SSL terminating outside of CF if we control the deployed on a subdomain. 

This helps us avoid the night mares of a 301 redirect during initial setup, were maybe testing stuff with a LE cert and not necessary having deployed our origin certs.

In short, 301s are bad when you are uncertain of what you may end up doing to a config, in this case a subdomain that we are not sure we may be needing in a certain way.

**Comment by raisedadead** [comment]
Turns out I missed a few things! Sorry about the noise. 


PS. We need tests for our NGINX somehow. You do not discover things unless you test them out.

**Comment by raisedadead** [comment]
I think that should be it. Tested this on .dev

**Comment by raisedadead** [comment]
Chat has now been rolled over to its own config, and deployed as well. 

The only bit that remains is soft launch and then deploying this config so that both Chinese and Espnol resolve correctly.

**Comment by ojeytonwilliams** [comment]
> PS. We need tests for our NGINX somehow. You do not discover things unless you test them out.

Interesting, I'd not thought of testing the proxy itself.  Well, we could 

- spin up some 'upstreams' at various ports
- set up a host file to map the domains to the proxy 
- send requests to those domains and confirm that we're getting the right responses from the right upstreams.  

Something like that?

I'm not quite sure how we'd handle https, though.

File: sites-enabled/10-www.freecodecamp.org.conf (hunk 1 above)
[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply sites-enabled/10-www.freecodecamp.org.conf:131]: Okay, I think I get that.  In which case, shouldn't we be using 301 for this and 302 for chinese?  Maybe I'm still missing something, since this `server_name` is specifically `www.freecodecamp.org` not `*.freecodecamp.org` (for that I could see us wanting a 302).

**Comment by raisedadead** [comment]
Sorry I should have clarified, we need static testing. E2E is not a big concern to be honest. I have seen example in the wild where others have setup the CI to install NGINX and simply run the `nginx -t` to test the syntax is all good.

At the minute we have pitfalls where we have accidental duplications or a missing semi-colon, etc.

File: sites-enabled/10-www.freecodecamp.org.conf (hunk 1 above)
[the remark this answers is not in the document]
Comment (raisedadead) [reply sites-enabled/10-www.freecodecamp.org.conf:131]: Yeah if you visit the file for the last commit, it should be clear. BTW, a `.freecodecamp.org`, notice the dot, is a catch all.

[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply sites-enabled/10-www.freecodecamp.org.conf:131]: Thanks. I'm not sure how I missed the `.` I need more sleep/coffee.
